File: 02_Top-Level-Declarations/01_Class/java/Prog.java
// No top-level functions in Java: main must be a class method.
public final class Prog {

	// 'static' indicates that this member is not associated with an instance
	public static final void main() {

		// 'final var' here is similar to Kotlin 'val' with type inference
		final var rc = new RegularClass(); 

		rc.mrc();
	}

	// Official main method calling an alternative overload
	public static void main(String[] args) { main(); }
}

File: 02_Top-Level-Declarations/01_Class/kotlin/prog.kt
// classes are public (and final/closed) by default
class RegularClass {

	// methods are public (and final/closed) by default
	// return type inferred from implementation
	fun mrc() { println("RegularClass") }

}

// top-level functions are possible
fun main() {

	// val declares a constant
	// type is inferred from the initialization expression
	val rc = RegularClass()
	
	rc.mrc()
}
